lib/socket.ts

```diff
@@ -772,6 +769,9 @@ export class Socket<
    */
   _cleanup() {
     this.leaveAll();
+    this.nsp._remove(this);
+    this.client._remove(this);
+    this.connected = false;
```

Comment:
I don't think `this.client._remove(this);` and `this.connected = false;` are needed here, since the socket is not actually connected if it's rejected by the middleware. What do you think?

Reply:
I agree, I just thought it does no harm since `client.remove` has a guard check and no-ops if the socket is not actually connected, but yeah, let me move it out

Reply:
Done.